src/main/kotlin/uk/gov/dluhc/notificationsapi/mapper/NotRegisteredToVoteTemplatePreviewDtoMapper.kt

```diff
@@ -9,10 +9,14 @@ import uk.gov.dluhc.notificationsapi.dto.NotRegisteredToVoteTemplatePreviewDto
 import uk.gov.dluhc.notificationsapi.models.GenerateNotRegisteredToVoteTemplatePreviewRequest
 import uk.gov.dluhc.notificationsapi.models.NotRegisteredToVotePersonalisation
 import uk.gov.dluhc.notificationsapi.models.SourceType
+import java.time.LocalDate
 
 @Mapper(uses = [LanguageMapper::class, CommunicationChannelMapper::class, SourceTypeMapper::class])
 abstract class NotRegisteredToVoteTemplatePreviewDtoMapper {
 
+    @Autowired
+    protected lateinit var deadlineMapper: DeadlineMapper
```

Comment:
The `DeadlineMapperTest` with its test function `should map deadline string` looks strange, seeing that it doesn't test permutations (non-null, null) of the date as well as the string. It may be that this won't ever be encountered but if it ever can, it should be covered by a test.

Reply:
So the chain of events here is roughly, front-end form -> service API -> Notificatiosn API -> Notify.
Theres validation on both the front-end form and service APIs to throw errors if a date in null btu time string is present so is a case that isn't going to be encountered here